build-system/tasks/extension-helpers.js

```diff
@@ -507,6 +511,46 @@ async function buildExtension(
   await buildExtensionJs(extDir, name, version, latestVersion, options);
 }
 
+/** @type {TransformCache} */
+let cssCache;
+
+/**
+ * Writes an extensions's CSS to its npm dist folder.
+ *
+ * @param {string} extDir
+ * @param {Object} options
+ * @return {Promise<void>}
+ */
+async function buildNpmCss(extDir, options) {
+  const startCssTime = Date.now();
+  const jssFile = (await globby(path.join(extDir, '**', '*.jss.js')))[0];
+  if (!jssFile) {
+    return;
+  }
+
+  // Lazily instantiate the TransformCache
+  if (!cssCache) {
+    cssCache = new TransformCache('jss-cache', '.css');
+  }
+
+  const {contents, hash} = await batchedRead(jssFile);
+  let css = await cssCache.get(hash);
+  if (!css) {
+    const options = {css: 'REPLACED_BY_BABEL_PLUGIN'};
+    await babel.transform(contents, {
+      filename: jssFile,
+      plugins: [
```

Comment:
Use callername, and configure this through the regular `build-system/babel-config` files.

Reply:
If I use `callerName`, how can I pass in the options object?

Reply:
Ooh, thank you Justin for pointing out what is a pretty important maintainability fix. @samouri, here's a rough recipe':

1. Move this config object to `build-system/babel-config/jss-config.js` and have it export `getJssConfig()`.
2. Plumb that config object through `./babel.config.js` and add an entry to the caller map under the key `'jss'`.
3. In this file, do the transform using the caller name. Babel will automatically fetch (and cache) the options object.

```js
// Code looks something like this. You'll need to plumb in the `filename` option as well.
const babelOptions = babel.loadOptions({caller: {name: 'jss'}});
const contents = // read contents from file
const result = babel.transformAsync(contents, babelOptions);
// add result to cache, await result, etc.
```

Reply:
Two downsides:
1. `babel.loadOptions` is slow. It will add ~250ms to what is currently taking closer to 10ms
2. We'll need to retrieve the value via `babelOptions.plugins[0][1].css`, which is not great

Reply:
> I didn't mean the babel options object. I meant the specific plugin's options object.

You're right, I misunderstood your original question 😃

@jridgewell Can you think of a more elegant way to extract the plugin from the options object?

Reply:
> Retrieval of the options object is brittle, and dependent on babel privates.

We can add an export in the config to simplify this.

```js
// babel-config/jss.js
export css = {};
// plugins: [ [ 'jss-plugin', { css: css } ] ]


// extension-helpers.js 
import { css } from 'babel-config/jss.js';
```

> babel.loadOptions has variable speed. on a cold cache will add ~800ms to what usually only takes ~100ms.

This is optimizing an already notoriously slow process. I much rather have centralized location for configs than the speed here.

Reply:
I guess its the best we can do. Thank you @jridgewell  / @rsimha for the help!